language/diem-vm/parallel-executor/src/executor.rs

```diff
@@ -113,8 +150,18 @@ where
                         }
 
                         // Process the output of a transaction
+                        let view = MVHashMapView {
+                            map: &versioned_data_cache,
+                            version: idx,
+                            scheduler: &scheduler,
+                            has_unexpected_read: AtomicBool::new(false),
+                        };
+                        let execute_result = task.execute_transaction(&view, txn);
+                        if view.has_unexpected_read() {
+                            continue;
```

Comment:
- A comment seems useful here; intuitively I'm not sure why it's safe to continue in this case
- If it's possible, perhaps we should log something here?

Reply:
So the idea is following: when there'e an unexpected write, we've already added this transaction back to the scheduler queue in the `MVHashMapView`, which is a bit unintuitive. Thus there's nothing we can do over there.

In terms of logging, I think we should log immediately when there's an unestimated read, which I should implement shortly.

Reply:
Got it, that makes sense + would be good to explain in the comment!